src/Data/Array/Mutable/Linear.hs

```diff
@@ -124,16 +131,17 @@ read = Unsafe.toLinear readUnsafe
           in  (arr, Unrestricted a)
       | otherwise = error "Read index out of bounds."
 
--- | Using first element as seed, resize to a constant array
-resize :: HasCallStack => Int -> Array a #-> Array a
-resize newSize (Array _ mutArr) =
-  let !(# a #) = Unsafe.readMutArr mutArr 0
-  in  Array newSize (Unsafe.newMutArr newSize a)
-
--- | Resize to a new constant array given a seed value
-resizeSeed :: HasCallStack => Int -> a -> Array a #-> Array a
-resizeSeed newSize seed (Array _ _) =
-  Array newSize (Unsafe.newMutArr newSize seed)
+-- | Resize an array. That is, given an array, a target size, and a seed
+-- value; resize the array to the given size using the seed value to fill
+-- in the new cells when necessary and copying over all the unchanged cells.
+--
+-- let b = resize n x a,
+--   then length b = n,
+--   and b[i] = a[i] for i < length a,
+--   and b[i] = x for length a <= i < n.
```

Comment:
I don't think that the formatting will be preserved by Haddock.

Reply:
Thanks, I will wrap it with a code block.